x-pack/plugins/ml/public/application/components/annotations/annotation_flyout/index.tsx

```diff
@@ -145,21 +159,21 @@ class AnnotationFlyoutUI extends Component<CommonProps & Props> {
   public validateAnnotationText = () => {
     // Validates the entered text, returning an array of error messages
     // for display in the form. An empty array is returned if the text is valid.
-    const { annotation } = this.props;
+    const { annotationState } = this.state;
     const errors: string[] = [];
-    if (annotation === null) {
+    if (annotationState === null) {
       return errors;
     }
 
-    if (annotation.annotation.trim().length === 0) {
+    if (annotationState.annotation.trim().length === 0) {
       errors.push(
         i18n.translate('xpack.ml.timeSeriesExplorer.annotationFlyout.noAnnotationTextError', {
```

Comment:
Should we remove this `.timeSeriesExplorer` because this annotation flyout is used in other places as well?

Reply:
We probably should! But I decided to avoid changing anything not related to this particular performance issue.